modules/metagetta/src/cljdoc_analyzer/metagetta/clojurescript.clj

```diff
@@ -132,22 +132,25 @@
      (comp/with-core-cljs state nil #(ana/analyze-file state file nil))))
   state)
 
-(defn- read-file [state source-path file exception-handler]
-  (try
-    (let [source  (io/file source-path file)
-          ns-name (:ns (ana/parse-ns source))
-          state   (analyze-file state source)]
-      (if-let [ns (ana/find-ns state ns-name)]
-        {ns-name
-         (-> ns
-             (select-keys [:name :doc])
-             (utils/update-some :doc utils/correct-indent)
-             (merge (-> ns-name meta (select-keys [:no-doc :skip-wiki :author :deprecated :added])))
-             (utils/remove-empties)
-             (assoc :publics (read-publics state ns-name source-path file)))}
-        (println "Dropping" file "because" ns-name "was not present in state. Is it missing an (ns) declaration?")))
-    (catch Exception e
-      (exception-handler e file))))
+(defn- load-source [state source-path ns-info exception-handler]
+  (let [file (::source-rel-path ns-info)
+        ns-name (:ns ns-info)
+        source (:source-file ns-info)]
+    (try
+      (let [state (analyze-file state source)]
```

Comment:
Is moving the try catch here just a refactor or does that impact anything?

Reply:
I think it is fine. It just let me reference the `file` in any thrown exception.
I suppose moving the other bindings up above the try is a bit questionable, but they are just data fetches from a map.